tests/elster_xml/common/test_xml_conversion.py

```diff
@@ -0,0 +1,293 @@
+from dataclasses import asdict, dataclass
+from typing import List
+
+import pytest
+
+from erica.elster_xml.common.xml_conversion import CustomDictParser, convert_object_to_xml
+
+
+class TestCustomDictParser:
+    def test_leaves_simple_object_unchanged(self):
```

Comment:
> Furthermore I am wondering if it would be enough if we only tested the CustomDictParser directly (it's init function). Because we then do not need to test the nested objects?

I thought about it, but I think this is way more graspable - at least I wouldn't be sure the integration works as expected. I really just implemented the init by try-and-erroring basically these tests. So I'd prefer it to stay that way.



> We could rename all those tests to follow the if_this_then_that convention. I find it quite hard to understand what is going on tbh

Sure.

Reply:
you test that with the other function anyway if I am correct? But I would be fine with that. However, we then might need to think about more test cases. At least I do not get when you use a nested object and when not.

Reply:
Let's talk about it? Because I'm not sure I get what you mean (read as: no idea what you're talking about. :D)